pkg/client/testclient/testclient.go

```diff
@@ -1,20 +1,75 @@
 package testclient
 
 import (
-	"k8s.io/kubernetes/pkg/apimachinery/registered"
+	"io/ioutil"
+
+	"k8s.io/kubernetes/pkg/api/meta"
+	kclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/fake"
 	kclient "k8s.io/kubernetes/pkg/client/unversioned"
-	"k8s.io/kubernetes/pkg/client/unversioned/testclient"
+	ktestclient "k8s.io/kubernetes/pkg/client/unversioned/testclient"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/yaml"
 
-	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/client"
+	projectapi "github.com/openshift/origin/pkg/project/api"
 )
 
 // NewFixtureClients returns mocks of the OpenShift and Kubernetes clients
-func NewFixtureClients(o testclient.ObjectRetriever) (osclient.Interface, kclient.Interface) {
-	oc := &Fake{}
-	oc.AddReactor("*", "*", testclient.ObjectReaction(o, registered.RESTMapper()))
-
-	kc := &testclient.Fake{}
-	kc.AddReactor("*", "*", testclient.ObjectReaction(o, registered.RESTMapper()))
+// with data populated from provided path.
+func NewFixtureClients(decoder runtime.Decoder, namespace, path string, objects ...runtime.Object) (client.Interface, kclientset.Interface, kclient.Interface, error) {
```

Comment:
this is a weird function now... it reads files from disk, it defaults namespaces... seems like three functions mixed up in one